packages/rum-core/src/domain/assembly.ts

```diff
@@ -82,6 +82,7 @@ export function startRumAssembly(
         }
 
         if (!isEmptyObject(commonContext.user)) {
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```

Comment:
Oh, this is a bug in the ESLint rule :) the type assertion is necessary to drop the "readonly". 👍